worker/registry.ts

```diff
@@ -88,22 +90,48 @@ export async function handleRegistryRequest(url: URL): Promise<Response> {
   }
   const remoteUrl = getBackingURL(module, version, path);
   const resp2 = await fetchSource(remoteUrl);
-
-  // JSX and TSX content type fix
+  let respText = await resp2.text();
   if (
-    remoteUrl.endsWith(".jsx") &&
-    !resp2.headers.get("content-type")?.includes("javascript")
+    headers.get("accept")?.includes("application/typescript") ||
+    headers.get("user-agent")?.includes("Deno/")
   ) {
-    resp2.headers.set("content-type", "application/javascript");
+    // JSX and TSX content type fix
+    if (
+      remoteUrl.endsWith(".jsx") &&
+      !resp2.headers.get("content-type")?.includes("javascript")
```

Comment:
It seems that `resp2.headers.get("content-type")` is repeated multiple times. Why don't you store it in a variable?